### PR TITLE
[SCAL-77792] - Doc UI changes

### PR DESCRIPTION
**Pull request by harsh-ts**
Ticket link -> https://thoughtspot.atlassian.net/browse/SCAL-77792

This PR contains:

- Added 40px margin-bottom to left nav components
- Changed padding for document and docmap to 24px
- Removed margin-top from h1 tag in document
- Added margin-bottom 40px to doc content
- Added font-weight-normal to links in docmap
- Active link functionality on right nav.

### CODE REVIEW

File: docs/src/components/Docmap/index.scss [hunk 1]
```diff
@@ -11,7 +12,7 @@
     overflow-y: auto;
 
     .tocTitle {
-        padding-left: 40px;
+        // padding-left: 40px;
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.scss:15]: Remove if not required

File: docs/src/components/Docmap/index.scss [hunk 2]
```diff
@@ -50,6 +50,10 @@
     display: inline-block;
 }
 
+.activeTag {
+    color: $secondarycolor !important;
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.scss:54]: Can we avoid !important? Are we trying to override any existing styles here?

File: docs/src/components/Docmap/index.tsx [hunk 3]
```diff
@@ -10,13 +10,32 @@ const Docmap = (props: { docContent: string }) => {
...
+            }
             setToc(tocEl.innerHTML);
         } else {
             setToc('');
         }
-    }, [props.docContent]);
+    }, [props.docContent, location.hash]);
+
+    const toggleActiveClass = (toc: Element, href: string) => {
+        toc.querySelectorAll("a").forEach((tag, index) => {
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.tsx:26]: Use single quotes, i.e., 'a' for consistency

File: docs/src/components/Docmap/index.tsx [hunk 4]
```diff
@@ -10,13 +10,32 @@ const Docmap = (props: { docContent: string }) => {
...
+            const temp = tag;
+            if (tag.getAttribute('href') === href) {
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.tsx:28]: What are we trying to do here, highlighting the currently selected page in the navigation?

File: docs/src/components/Docmap/index.tsx [hunk 5]
```diff
@@ -10,13 +10,32 @@ const Docmap = (props: { docContent: string }) => {
...
+                temp.classList.add('activeTag');
+            } else {
+                if (tag.classList.contains('activeTag')) {
+                    temp.classList.remove('activeTag');
+                }
+            }
+            toc.querySelectorAll("a")[index].innerHTML = temp.innerHTML;
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.tsx:35]: Single quotes

File: docs/src/components/Document/index.scss [hunk 6]
```diff
@@ -2,13 +2,18 @@
 
 .documentWrapper {
     width: calc(100% - 300px);
-    padding: 60px;
+    padding: 24px;
```
Comment (ayonghosh) [inline docs/src/components/Document/index.scss:5]: I see this padding used multiple times. Let's create a mixin for this. Also remove the next commented out line if not required.

File: docs/src/components/Docmap/index.scss (hunk 1 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.scss:15]: My mistake, I haven't pushed the latest commit where I have fixed this

File: docs/src/components/Docmap/index.tsx (hunk 3 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.tsx:26]: Will use single quote as standard

File: docs/src/components/Docmap/index.tsx (hunk 4 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.tsx:28]: Yes, we are doing it by sung above logic

File: docs/src/components/Docmap/index.scss (hunk 1 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.scss:15]: Yeah sure

File: docs/src/components/Docmap/index.tsx (hunk 3 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.tsx:26]: Sure.

File: docs/src/components/Document/index.scss (hunk 6 above)
Comment (harsh-ts) [reply docs/src/components/Document/index.scss:5]: Yeah I'll do that.

File: docs/src/components/Docmap/index.scss (hunk 2 above)
Comment (harsh-ts) [reply docs/src/components/Docmap/index.scss:54]: That is required. Otherwise we'll have to write this class in multiple places.

**Review by ayonghosh — APPROVED** [review]

**Pull request merged**